src/cli/parse-flags.ts

```diff
@@ -161,6 +164,47 @@ const parseNumberArg = (flags: ConfigFlags, args: string[], configCaseName: Numb
   }
 };
 
+/**
+ * Parse a CLI argument which may be either a string or a number
+ *
+ * @param flags the config flags object, while we'll modify
+ * @param args our CLI arguments
+ * @param configCaseName the argument we want to look at right now
+ */
+const parseStringNumberArg = (flags: ConfigFlags, args: string[], configCaseName: StringNumberCLIArg) => {
+  if (!['number', 'string'].includes(typeof flags[configCaseName])) {
+    flags[configCaseName] = null;
+  }
+
+  const { value, matchingArg } = getValue(args, configCaseName);
+
+  if (value !== undefined && matchingArg !== undefined) {
+    if (CLI_ARG_STRING_REGEX.test(value)) {
+      // if it matches the regex we treat it like a string
+      flags[configCaseName] = value;
+    } else {
+      // it was a number, great!
+      flags[configCaseName] = parseInt(value, 10);
+    }
+    flags.knownArgs!.push(matchingArg);
+    flags.knownArgs!.push(value);
+  }
+};
+
+/**
+ * We use this regular expression to detect CLI parameters which
+ * should be parsed as string values (as opposed to numbers) for
+ * the argument types for which we support both a string and a
+ * number value.
+ *
+ * The regex tests for the presence of at least one character which is
+ * _not_ a digit (`\d`) or a period (`\.`). Thus we'll match a string
+ * like `"50%"`, but not a string like `"50"` or `"5.0"`. If it matches
+ * a given string we conclude that the string should be parsed as a
+ * string literal, rather than using `parseInt` to convert it to a number.
+ */
+const CLI_ARG_STRING_REGEX = /[^\d\.]+/g;
```

Comment:
I think this should be pretty solid, and I don't think we need to support negative numbers, separators like `_` or `,`, etc. But always good to be sure with regexes!

Reply:
ooo I forgot that JS had scientific notation support! will add that and a test case for it